editor/animation_track_editor.cpp

```diff
@@ -4100,107 +4100,43 @@ void AnimationTrackEditor::insert_node_value_key(Node *p_node, const String &p_p
 		return;
 	}
 	InsertData id;
-	id.path = np;
+	id.path = path;
 	id.track_idx = -1;
-	id.value = p_value;
+	id.value = value;
 	id.type = Animation::TYPE_VALUE;
 	id.query = vformat(TTR("property '%s'"), p_property);
-	id.advance = false;
+	id.advance = p_advance;
 	// Dialog insert.
 	_query_insert(id);
 }
 
-void AnimationTrackEditor::insert_value_key(const String &p_property, const Variant &p_value, bool p_advance) {
+void AnimationTrackEditor::insert_value_key(const String &p_property, bool p_advance) {
 	EditorSelectionHistory *history = EditorNode::get_singleton()->get_editor_selection_history();
 
 	ERR_FAIL_NULL(root);
 	ERR_FAIL_COND(history->get_path_size() == 0);
 	Object *obj = ObjectDB::get_instance(history->get_path_object(0));
-	ERR_FAIL_NULL(Object::cast_to<Node>(obj));
-
-	// Let's build a node path.
-	Node *node = Object::cast_to<Node>(obj);
-	String path = root->get_path_to(node, true);
-
-	if (Object::cast_to<AnimationPlayer>(node) && p_property == "current_animation") {
-		if (node == AnimationPlayerEditor::get_singleton()->get_player()) {
-			EditorNode::get_singleton()->show_warning(TTR("AnimationPlayer can't animate itself, only other players."));
-			return;
-		}
-		_insert_animation_key(path, p_value);
-		return;
-	}
-
-	for (int i = 1; i < history->get_path_size(); i++) {
-		String prop = history->get_path_property(i);
-		ERR_FAIL_COND(prop.is_empty());
-		path += ":" + prop;
-	}
-
-	path += ":" + p_property;
-
-	NodePath np = path;
-
-	// Locate track.
 
-	bool inserted = false;
+	MultiNodeEdit *multi_node_edit = Object::cast_to<MultiNodeEdit>(obj);
+	if (multi_node_edit != nullptr) {
```

Comment:
MultiNodeEdit is RefCounted, so this is a more standard way of handling it:
```suggestion
	Ref<MultiNodeEdit> multi_node_edit = obj;
	if (multi_node_edit.is_valid()) {
```

Reply:
That does seem to work, by virtue of `RefCounted::operator=(const Variant &)`, causing it to implicitly cast the pointer to a `Variant` first. I went ahead and made the suggested change.

Reply:
Looks like it doesn't work. You can try `Ref<MultiNodeEdit> multi_node_edit(obj)` or `Ref<MultiNodeEdit> multi_node_edit = Ref<MultiNodeEdit>(obj);`.

Reply:
It's somewhat troubling that MSVC was happy to do the implicit `Variant` conversion but GCC was not. I wonder if it's because Godot isn't providing [`/permissive-`](https://learn.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-170) when compiling with MSVC. Your latest suggestion seems to work in GCC as well though.